gnd/src/main/java/com/google/android/gnd/persistence/local/room/entity/FeatureEntity.java

```diff
@@ -134,10 +168,33 @@ public static Feature toFeature(FeatureEntity featureEntity, Project project) {
       return builder.build();
     }
 
+    if (featureEntity.getPolygonVertices() != null) {
+      PolygonFeature.Builder builder =
+          PolygonFeature.builder().setVertices(stringToList(featureEntity.getPolygonVertices()));
+      fillFeature(builder, featureEntity, project);
+      return builder.build();
+    }
+
     throw new LocalDataConsistencyException(
         "No geometry data found in feature " + featureEntity.getId());
   }
 
+  public static ImmutableList<Point> stringToList(String vertices) {
+    if (vertices == null) {
+      Timber.d("vertices are null");
+      return null;
```

Comment:
This method isn't declared as `@Nullable`. returning `ImmutableList.of()` might be more fault resistant.